[ICDS Dashboard][fix]: Tabs remains consistent while drilling down from the P.S. to AWC report

ShubhamBansal1997 opened the pull request:
- JIRA Issue:
https://dimagi-dev.atlassian.net/browse/ICDS-1097

##### SUMMARY
As of now if someone goes down from the Program Summary page to the AWC report, it always lands up on the PSE tab of the AWC report although if the person selected any other tab like demographics on Program Summary Page. After applying the fix the mapping would be like this.

- maternal_child of PS -> maternal_child of awc report
- cas_Reach of PS -> pse of awc_report
- demographics of PS -> demographics of awc_reports
- awc infra of PS -> awc infra of awc_report

ShubhamBansal1997 committed:
[fix]: Tabs remains consistent while drilling down from the program summary page to awc report page

Right Now:
- All the tabs -> pse of awc_report
After Bug Fix:
- maternal_child of PS -> maternal_child of awc report
- cas_Reach of PS -> pse of awc_report
- demographics of PS -> demographics of awc_reports
- awc infra of PS -> awc infra of awc_report

- Github Issue:

- JIRA Issue:
https://dimagi-dev.atlassian.net/browse/ICDS-1097

Authored-by: Shubham Bansal <shubhambansal17@hotmail.com>
Signed-off-by: Shubham Bansal <shubhambansal17@hotmail.com>

czue commented on custom/icds_reports/static/js/directives/map-or-sector-view/map-or-sector-view.directive.js:
```diff
@@ -104,10 +104,21 @@ function MapOrSectorController($location, storageService, locationsService) {
+                            } else if ($location.path().indexOf('icds_cas_reach') !== -1) {
+                                awcReportPath += '/pse';
+                            } else {
+                                awcReportPath += '/pse';
```
nit: I think it would make more sense to keep the default simply `'awc_reports'` so that if it changes later that would get handled by the route rule and not duplicately here.

ShubhamBansal1997 replied:
Yeah that would be much better, updated accordingly.

czue commented on custom/icds_reports/static/js/directives/map-or-sector-view/map-or-sector-view.directive.js:
```diff
@@ -104,10 +104,21 @@ function MapOrSectorController($location, storageService, locationsService) {
                         if (location.location_type_name === 'awc') {
-                            $location.path('awc_reports');
+                            var awcReportPath = 'awc_reports';
+                            if ($location.path().indexOf('maternal_child') !== -1 || $location.path().indexOf('maternal_and_child') !== -1) {
```
Not blocking, but I think it's possible this would be a little more comprehensible (and be slightly more DRY) if you defined a data structure like this:

```javascript
{
  'maternal_child': 'maternal_child',
  'maternal_and_child': 'maternal_child',
  'demographics': 'demographics',
  // etc.
}
```

And then updated the path in a loop.

ShubhamBansal1997 replied:
Updated, Please review

czue reviewed (commented):
Congrats on the first PR and generally looks good! :clap: :clap: 

I think only the feedback on the default value is semi-blocking (unless you disagree with it).

czue commented:
Also you should add the appropriate product label for this (which is why the CI is failing): https://confluence.dimagi.com/display/ccinternal/Tagging+PRs+with+Product+labels

All ICDS Dashboard work is likely to be custom.

ShubhamBansal1997 committed:
[refactor]: Use of DS approach instead of the if else to identify the
particular route for a tab

- Github Issue:

Authored-by: Shubham Bansal <shubhambansal17@hotmail.com>
Signed-off-by: Shubham Bansal <shubhambansal17@hotmail.com>

ShubhamBansal1997 committed:
[style]: resolved code style issues arised by stickler-ci bot

- Github Issue:

Authored-by: Shubham Bansal <shubhambansal17@hotmail.com>
Signed-off-by: Shubham Bansal <shubhambansal17@hotmail.com>

czue commented on custom/icds_reports/static/js/directives/map-or-sector-view/map-or-sector-view.directive.js:
```diff
@@ -104,10 +104,28 @@ function MapOrSectorController($location, storageService, locationsService) {
+                                }
+                            }
+                            // PSE is the deafult tab
+                            if (awcReportPath === 'awc_reports') {
```
I don't think you need this if statement do you? if the default is `'awc_reports'` then [this line](https://github.com/dimagi/commcare-hq/blob/master/custom/icds_reports/static/js/icds_dashboard_utils.js#L118-L120) will handle it (this is basically the same comment I already made)

czue reviewed (commented):
Looking good, just one more small thing

ShubhamBansal1997 committed:
[revert]: removed the default pse route statement as it is maintained by
the routers https://github.com/dimagi/commcare-hq/blob/master/custom/icds_reports/static/js/icds_dashboard_utils.js#L118-L120

- Github Issue:

Authored-by: Shubham Bansal <shubhambansal17@hotmail.com>
Signed-off-by: Shubham Bansal <shubhambansal17@hotmail.com>

Rohit25negi reviewed (approved):
yay! first PR.. 🎊 
Looks good to me.

czue commented:
:tada:
